### PR TITLE
Add partiql-planner and partiql-parser Maven publishing plugin v0.12.1

### PR DESCRIPTION
**Pull request by alancai98**
## Description
Adds a Maven publishing step to `partiql-planner` and `partiql-parser` and changes partiql-lang-kotlin's dependency on `partiql-planner` to be `api`.

## Other Information
- Updated Unreleased Section in CHANGELOG: Yes

- Any backward-incompatible changes? No

- Any new external dependencies? No

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? Yes

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-parser/build.gradle.kts [hunk 1]
```diff
@@ -51,8 +51,8 @@ tasks.processResources {
     }
 }
 
-// publish {
-//     artifactId = "partiql-parser"
-//     name = "PartiQL Parser"
-//     description = "PartiQL's Parser"
-// }
```
Comment (alancai98) [inline partiql-parser/build.gradle.kts:58]: Need to check on reason this was commented out initially

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,14 @@ Thank you to all who have contributed!
 
 -->
 
+## [0.12.1] - 2023-09-19
+
+### Added
```
Comment (johnedquinn) [inline CHANGELOG.md:30]: Same as other PRs.

**Review by johnedquinn — COMMENTED** [review]
Why the deviation for the target branch (`v0.12.1-partiql-planner-publish`)?

**Comment by alancai98** [comment]
> Why the deviation for the target branch (`v0.12.1-partiql-planner-publish`)?

I need to sync w/ @yliuuuu on this. I saw the [v0.12.1 branch](https://github.com/partiql/partiql-lang-kotlin/tree/v0.12.1 was created previously in https://github.com/partiql/partiql-lang-kotlin/pull/1200. But it was later reverted?

I decided to just create a new branch called `v0.12.1-partiql-planner-publish`. If we decide that the v0.12.1 branch was previously created in error, I can 
1. delete the old v0.12.1
2. push a new v0.12.1 (same as v0.12.0)
3. make this PR target the new v0.12.1

**Comment by alancai98** [comment]
PR should now target [v0.12.1 branch](https://github.com/partiql/partiql-lang-kotlin/tree/v0.12.1) which was created off of v0.12.0.

File: partiql-parser/build.gradle.kts [hunk 3]
```diff
@@ -51,8 +51,8 @@ tasks.processResources {
...
+publish {
+    artifactId = "partiql-parser"
+    name = "PartiQL Parser"
+    description = "PartiQL's Parser"
```
Comment (johnedquinn) [inline partiql-parser/build.gradle.kts:57]:
```suggestion
    description = "PartiQL's experimental Parser"
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -25,6 +25,14 @@ Thank you to all who have contributed!
 
 -->
 
+## [0.12.1] - 2023-09-19
+
+### Fixed
+- Fixes build failure for version `0.12.0` by publishing `partiql-plan` and `partiql-parser` as an independent artifact. Please note that `partiql-plan` is experimental.
```
Comment (johnedquinn) [inline CHANGELOG.md:31]:
```suggestion
- Fixes build failure for version `0.12.0` by publishing `partiql-plan` and `partiql-parser` as independent artifacts. Please note that `partiql-plan` and `partiql-parser` are experimental.
```

**Review by johnedquinn — CHANGES_REQUESTED** [review]
Only small comment is that our official production parser is still under `org.partiql.lang.syntax.Parser` -- and the experimental parser is in `org.partiql.parser`.

The CHANGELOG might indicate to some users that they should onboard using the experimental package.

**Comment by yliuuuu** [comment]
> > Why the deviation for the target branch (`v0.12.1-partiql-planner-publish`)?
> 
> I need to sync w/ @yliuuuu on this. I saw the [v0.12.1 branch](https://github.com/partiql/partiql-lang-kotlin/tree/v0.12.1) was created previously in #1200. But it was later reverted?
> 
> I decided to just create a new branch called `v0.12.1-partiql-planner-publish`. If we decide that the v0.12.1 branch was previously created in error, I can
> 
> 1. delete the old v0.12.1
> 2. push a new v0.12.1 (same as v0.12.0)
> 3. make this PR target the new v0.12.1

Can confirm. The old v.0.12.1 was created in error. Sorry should have clean that up.

**Review by johnedquinn — APPROVED** [review]

**Pull request merged**